### DISCUSSION TITLE
`ui.button()` path

### DISCUSSION THREAD

**Discussion by azim-b** [h2oai/wave#1497] [Q&A] [answered]
`path` related questions for a button:

- currently, the path is opened in a new tab. Could we make this more flexible so that we can open the link in the same tab (ex: location hash is `x`, and we want to retry the process on this page, so the button retry's path = `#x`)
- even if `path` is specified, the api wants a `name`, even though `name` will be ignored. Is this expected?
- can we extend this `path` functionality to a `ui.mini_button()`?

**Comment by mturoci** [comment, marked as answer]
> 1

Path is used for external links (starting with `http://...`). Hash routing is still handled by `name` attr.

> 2

Yes, this is expected, there is no other way to do it.

> 3

Yes, this is doable, please file a feature request.

**Reply by azim-b** [reply]
1. I was asking because I was wondering whether we can "refresh" the page with the same location. Today, this is not possible, as the button click is not handled if the button's name attribute matches the current location hash. However, I remember seeing a conversation where the possibility of adding this was discussed. So, I will mark this as resolved.
